dags/dsn_data_preparation.py

```diff
@@ -1,23 +1,92 @@
-import os
+import re
+import unicodedata
 from pathlib import Path
 
+import pandas as pd
 from airflow import DAG
 from airflow.decorators import task
 from airflow.models import Variable
 
-from dags.common import default_dag_args, s3
+from dags.common import db, default_dag_args, s3
 
 
 DAG_ID = "dsn_data_preparation"
 LOCAL_TMP_ROOT = Path("/tmp") / DAG_ID
 S3_PREFIX_ROOT = "depot_dsn/raw"
+RAW_SCHEMA = "raw_dsn"
 
 
-def _get_s3_config():
-    return {
-        "bucket": Variable.get("DATASTORE_S3_PILOTAGE_BUCKET_NAME"),
-        "prefix_root": S3_PREFIX_ROOT,
-    }
+FILE_TABLE_MAPPING = {
+    "01. Critère Géographique et Sexe.xlsx": "dsn_geographie_et_sexe",
+    "02. Critère Géographique et Nature de Contrat.xlsx": "dsn_geographie_et_nature_contrat",
+    "03. Critère Géographique et Secteur d'Activité.xlsx": "dsn_geographie_et_secteur_activite",
+    "04. Type de SIAE et Sexe.xlsx": "dsn_type_siae_et_sexe",
+    "05. Type de SIAE et Nature de Contrat.xlsx": "dsn_type_siae_et_nature_contrat",
+    "06. Type de SIAE et Secteur d'Activité.xlsx": "dsn_type_siae_et_secteur_activite",
+    "07. Type de SIAE et Critère Géographique.xlsx": "dsn_type_siae_et_geographie",
+}
+
+
+def _normalize_col_name(col) -> str:
+    s = str(col)
+
+    # 1) quitar acentos
+    s = unicodedata.normalize("NFKD", s)
+    s = "".join(c for c in s if not unicodedata.combining(c))
+
+    # 2) normalizar comillas y bajar a minúsculas
+    s = s.replace("’", "'")
+    s = s.strip().lower()
+
+    # 3) quitar la preposición francesa "d'" como palabra
+    #    ejemplos:
+    #      "secteur d'activite" -> "secteur activite"
+    #      "d'emploi"           -> "emploi"
+    s = s.replace(" d'", " ")
+    if s.startswith("d'"):
+        s = s[2:].lstrip()
+
+    # 4) quitar cualquier apóstrofe que quede
+    s = s.replace("'", "")
+
+    # 5) espacios -> guiones bajos, colapsar múltiples
+    parts = s.split()
+    s = "_".join(parts)
+    while "__" in s:
+        s = s.replace("__", "_")
+
+    # 6) quitar guiones bajos al principio y al final
```

Comment:
les commentaires en général : bof
les commentaires qui expliquent ce que fait le code : bof,[ on préfère quand ça dit pourquoi](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/)
les commentaires en espagnol : vraiment pas ^^

et sinon je ne suis pas super fan de cette méthode qui me semble etre une normalisation pas vraiment "officielle" : pourquoi "d'" et pas "de" , pourquoi ne pas virer dans ce cas toutes les " à ", "entre" , ... Quel est l'objectif ?

Sinon juste utiliser `slugify` ?